### PR TITLE
changed file structure and lots of other changes

### PR DESCRIPTION
**Pull request by Ngope**
restructured the app to make it better XD
made new endpoints for testing spotify api calls
upgraded to use node 20
changed all calls to use axios

yall should hit the test endpoints :) pretty cool

### CODE REVIEW

File: src/app/routes/auth.ts [hunk 1]
```diff
@@ -98,7 +123,8 @@ router.get("/callback", async function (req: Request, res: Response) {
     try {
       // Send the POST request using Axios
       const response = await axios(authOptions);
-
+      config.accessToken = response.data.access_token;
+      config.refreshToken = response.data.refresh_token;
       // For demonstration, let's send a response back to the client with a success message
```
Comment (peetngo) [inline src/app/routes/auth.ts:128]: can you remove

File: src/app/routes/profile.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import config from "../../config/config"
+
+const router = express.Router();
+
+// Define routes
+router.get('/', getProfile);
+
+//define function
+async function getProfile (req: Request, res: Response) {
+    // config.accessToken = "BQB9HKYpA27UUAxviPPL5_0dL0iIu2Lp6K3f7WCVBRmitMeGP7FcbbgV1CzMgHwVS8hVDv58rcYRBAPHGldb-DehtZTwgElnad1YBDFF6pnWuX_HhnGNEEcP4ukOHF5MwKYqqOwlu2YsGW7WTZBVdpujK0_tji1MijLisYfBCLZXYIZsccpFOEOkrfO4Z5ToWtJsBMoZxHCKv7YZeMDfFQ"
```
Comment (peetngo) [inline src/app/routes/profile.ts:12]: do we need this

File: src/app/routes/test.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+router.get('/', emoji);
+router.get('/peter', peter);
+router.get('/blake', balake);
+
+//define function
+async function emoji (req: Request, res: Response) {
+    return res.status(200).json(['😀', '😳', '🙄'])
+}
+async function peter (req: Request, res: Response) {
+    console.log("I am a lil bitch XDXD -peter")
```
Comment (peetngo) [inline src/app/routes/test.ts:15]: 👎

File: src/app/routes/auth.ts [hunk 4]
```diff
@@ -45,28 +47,51 @@ async function getProfile(accessToken: string | null) {
...
 }
 
+router.get("/refreshTokens", async function (req: Request, res: Response) {
+  const authOptions = {
+    url: "https://accounts.spotify.com/api/token",
+    method: "POST", // Ensure the method is set to POST
+    data: {
+      client_id: client_id,
+      refresh_token: config.refreshToken,
+      grant_type: 'refresh_token',
```
Comment (peetngo) [inline src/app/routes/auth.ts:78]: are these coming from .env/config?

**Review by Blake254 — APPROVED** [review]
looks clean

File: src/app/routes/profile.ts (hunk 2 above)
Comment (Ngope) [reply src/app/routes/profile.ts:12]: no but thats how i was testing my profile call

File: src/app/routes/auth.ts (hunk 4 above)
Comment (Ngope) [reply src/app/routes/auth.ts:78]: config since on you have to call login to get tokens

**Pull request merged**